File: src/dataset/javascript/secret-handshake_exercism-javascript-ffflorian/claude/ClaudeSonnet4_secret-handshake_v3.js
const availableCommands = ['jump', 'close your eyes', 'double blink', 'wink'];

export function commands(input) {
  const result = [];
  let shouldReverse = false;
  
  for (let bit = 1; bit < 16; bit <<= 1) {
    if (input & bit) {
      if (bit === 1) {
        shouldReverse = true;
      } else {
        const commandIndex = Math.log2(bit) - 1;
        result.push(availableCommands[commandIndex]);
      }
    }
  }
  
  return shouldReverse ? result.reverse() : result;
}

File: src/dataset/cpp/secret_handshake_Exercism_thefullarcticfox/claude/ClaudeSonnet4_secret_handshake_v3.cpp
#include "secret_handshake.h"
#include <algorithm>
using namespace std;

namespace secret_handshake {

	vector<string> commands(unsigned int bits) {
		static constexpr const char* actions[] = {
			"wink",
			"double blink", 
			"close your eyes",
			"jump"
		};
		
		vector<string> res;
		res.reserve(4);
		
		for (int i = 0; i < 4; ++i) {
			if (bits & (1u << i)) {
				res.emplace_back(actions[i]);
			}
		}
		
		if (bits & 0b10000u) {
			std::reverse(res.begin(), res.end());
		}
		
		return res;
	}
}  // namespace secret_handshake